src/interpreter/index.ts

```diff
@@ -1614,14 +1615,24 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
 
     evaluate(this: Interpreter, expression: Expr.Expression): BrsType {
         this.location = expression.location;
+        this.stack.push(this.location);
         this.reportCoverageHit(expression);
-        return expression.accept<BrsType>(this);
+
+        let value = expression.accept<BrsType>(this);
+
+        this.stack.pop();
```

Comment:
Why is the location immediately popped out from the stack? Wouldn't the stack be empty in all cases when calling getStackTrace?

Reply:
Great question! `expression.accept` and `statement.accept` work recursively -- they will evaluate the current expression/statement, which will then require evaluating whatever expressions/statements it contains, so on and so forth. For example:
```brs
sub main()          '  ==> 1. `accept`, which then evaluates the inside of this function (`Block`)
    a = [1]         '  ==> 2. `accept` the `Block`, which then evaluates/executes line-by-line, starting with this one.
    print a[0] + 1  '  ==> 3. `accept` the statement, which then `accept`s (evaluates) `a[0]`, steps out, then executes the entire statement
...
```

At each step, we haven't yet exited the relevant `execute`/`evaluate` call for that `statement`/`expression`. When we're done with a statement/expression and have moved on to the next one, then we exit the `accept` call and pop the location from the stack.

Reply:
Got it!, thanks for the detailed explanation.

Reply:
> > Great question! `expression.accept` and `statement.accept` work recursively -- they will evaluate the current expression/statement, which will then require evaluating whatever expressions/statements it contains, so on and so forth. For example:
> > 
> > ```brightscript
> > sub main()          '  ==> 1. `accept`, which then evaluates the inside of this function (`Block`)
> >     a = [1]         '  ==> 2. `accept` the `Block`, which then evaluates/executes line-by-line, starting with this one.
> >     print a[0] + 1  '  ==> 3. `accept` the statement, which then `accept`s (evaluates) `a[0]`, steps out, then executes the entire statement
> > ...
> > ```
> > 
> > At each step, we haven't yet exited the relevant `execute`/`evaluate` call for that `statement`/`expression`. When we're done with a statement/expression and have moved on to the next one, then we exit the `accept` call and pop the location from the stack.
> 
> Got it!, thanks for the detailed explanation. 

Excellent description!